packages/@lwc/babel-plugin-component/src/decorators/index.ts

```diff
@@ -1,38 +1,43 @@
 /*
- * Copyright (c) 2018, salesforce.com, inc.
+ * Copyright (c) 2023, salesforce.com, inc.
  * All rights reserved.
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-const moduleImports = require('@babel/helper-module-imports');
-const { DecoratorErrors } = require('@lwc/errors');
-
-const { DECORATOR_TYPES, LWC_PACKAGE_ALIAS, REGISTER_DECORATORS_ID } = require('../constants');
-const {
-    generateError,
-    isClassMethod,
-    isSetterClassMethod,
-    isGetterClassMethod,
-} = require('../utils');
-
-const api = require('./api');
-const wire = require('./wire');
-const track = require('./track');
+import { Node, types, Visitor } from '@babel/core';
+import { NodePath, Binding } from '@babel/traverse';
+import { addNamed } from '@babel/helper-module-imports';
+import { DecoratorErrors } from '@lwc/errors';
+import { DECORATOR_TYPES, LWC_PACKAGE_ALIAS, REGISTER_DECORATORS_ID } from '../constants';
+import { generateError, isClassMethod, isSetterClassMethod, isGetterClassMethod } from '../utils';
+import { BabelAPI, LwcBabelPluginPass } from '../types';
+import api from './api';
+import wire from './wire';
+import track from './track';
+import { ImportSpecifier } from './types';
 
 const DECORATOR_TRANSFORMS = [api, wire, track];
 const AVAILABLE_DECORATORS = DECORATOR_TRANSFORMS.map((transform) => transform.name).join(', ');
 
-function isLwcDecoratorName(name) {
+export type DecoratorMeta = {
+    name: 'api' | 'track' | 'wire';
+    propertyName: string;
+    path: NodePath<types.Decorator>;
+    decoratedNodeType: string;
+    type?: 'property' | 'getter' | 'setter' | 'method';
+};
+
+function isLwcDecoratorName(name: string) {
     return DECORATOR_TRANSFORMS.some((transform) => transform.name === name);
 }
 
 /** Returns a list of all the references to an identifier */
-function getReferences(identifier) {
-    return identifier.scope.getBinding(identifier.node.name).referencePaths;
+function getReferences(identifier: NodePath<types.StringLiteral>) {
```

Comment:
This looks weird. Should it be `identifier: NodePath<types.Identifier>`?

Reply:
Yes, this type seems more correct! I was able to get rid of the `as any` cast. Thanks!